Include peeled brain as a regular surface and persist in projects

AksharaSBhat opened the pull request:
### What does this PR do?
This PR ensures that the peeled brain is treated as a regular surface but restricts color changes.
It also fixes an issue where the peeled brain was not saved and restored when reopening a project.
### Changes made
Added peel attribute to Surface to track peeled surfaces.
Updated CreateSurfaceFromPolydata to properly assign peel=True.
Modified SavePlist and OpenPlist to persist peel state.
Disabled "Change Color" button for peeled surfaces in the UI.
Disabled `button_colour` for peel in `task_surface.py`
### How to test
Create a peeled brain surface.
Save and close InVesalius.
Reopen the project and check if the peeled brain is restored.
Right-click on the peeled brain → "Change Color" should be disabled.
### Closes #796

AksharaSBhat commented:
@vhosouza @rmatsuda @henrikkauppi
Hi! Please take a look at this PR when you get a chance. I'd appreciate any feedback or suggestions. Thanks!

rmatsuda commented:
Hi! Many thanks for the fixes. 

I found a bug: I tried to create a peel using a mask file, but it is not applying the transformations. It’s supposed to be overlapping each other. Please take a look at the brain mesh handler.
![image](https://github.com/user-attachments/assets/cbfb610a-c332-4fc8-a3d2-d970eebafe20)

Another thing, the opacity is too blurry now. Look at how it was before:
![image](https://github.com/user-attachments/assets/1f450e99-b70f-4828-9222-6f483bfe85bf)

rmatsuda commented:
Regarding performance, it takes a while to change the peeling depth. I also noticed that the peel blinks twice. Could you try to optimize it, please?

AksharaSBhat commented:
Hi, could you guide me as to how can I get a mask file for creating a peel surface. Regarding the opacity, due to a bug in my code, opacity is treated as transparency, I will fix it. Could you please check and verify if reducing opacity(reducing transparency) removes the blurriness. I will also look into blinking of surface while changing depth, I believe it is being re-rendered twice

AksharaSBhat commented:
I have fixed the issue with opacity and added back face culling for peeled surfaces to remove the blurriness. Also, it no longer re-renders twice, so it only blinks once and it is relatively quick. I extracted a mask_1.plist from the inv3 file and used it to generate a peel surface and it perfectly overlapped with existing 3D surface. Could you tell me how can I reproduce the issues you are facing?

AksharaSBhat commented:
Could you please look into this again and if issues persist, tell me how to reproduce the issue @rmatsuda @paulojamorim.  extracted a mask_1.plist from the inv3 file and used it to generate a peel surface and it perfectly overlapped with existing 3D surface. I have fixed the blurriness and blinking issue.
